src/battle_util.c

```diff
@@ -10855,14 +10855,21 @@ void TryRestoreHeldItems(void)
 {
     u32 i;
     u16 lostItem = ITEM_NONE;
-
+    bool32 returnNPCItems = B_RETURN_STOLEN_NPC_ITEMS >= GEN_5 && gBattleTypeFlags & BATTLE_TYPE_TRAINER;
+    
     for (i = 0; i < PARTY_SIZE; i++)
     {
-        if (B_RESTORE_HELD_BATTLE_ITEMS >= GEN_9 || gBattleStruct->itemLost[i].stolen)
+        // Check if held items should be restored after battle based on generation
+        if (B_RESTORE_HELD_BATTLE_ITEMS >= GEN_9 || gBattleStruct->itemLost[i].stolen || returnNPCItems)
         {
             lostItem = gBattleStruct->itemLost[i].originalItem;
-            if (lostItem != ITEM_NONE && ItemId_GetPocket(lostItem) != POCKET_BERRIES)
-                SetMonData(&gPlayerParty[i], MON_DATA_HELD_ITEM, &lostItem);  // Restore stolen non-berry items
+
+            // Check if the lost item should be restored
+            if ((lostItem != ITEM_NONE || returnNPCItems)
+                && ItemId_GetPocket(lostItem) != POCKET_BERRIES)
+            {
+                SetMonData(&gPlayerParty[i], MON_DATA_HELD_ITEM, &lostItem);
+            }
```

Comment:
I think the if statement is small enough for one line and we don't use brackets for a one line unless it is in an if-else-block that has ifs(else) blocks with multiply lines. 

Also the NPC item is still stolen if the user used up a berry in the battle.

Reply:
mon triggers it's own berry during battle, then uses covet to steal opponents held item (a potion in this case). After the battle ends the mon still holds the potion.